app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/auth/EnrolmentExtractor.scala

```diff
@@ -36,16 +39,11 @@ trait EnrolmentExtractor {
       )
 
   def enrolledForService(loggedInUser: LoggedInUserWithEnrolments, service: Service): Option[Eori] = service match {
-    case Service.ATaR => enrolledATar(loggedInUser)
+    case Service.ATaR => identifierFor(ATaR.enrolmentKey, EoriIdentifier, loggedInUser).map(Eori)
```

Comment:
Why do we need the "match"?   `service.enrolmentKey` should do the job.

Reply:
Good point, I changed this.

Additionally I added tests for the whole trait because it didn't have any tests (except the places where it has been used)

Reply:
Just wondering if every enrolment will have a EORI identifier, but I'm 99% sure that this is the pattern that all services that we will be supporting will need to have an EORI in the enrolment. It's not UTR or NINO